app/llm/performance_enhanced.py

```
@@ -0,0 +1,142 @@
+import time
+import logging
+import threading
+from typing import Any, Dict, Optional
```

Ruff (0.8.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)